screens/visualization/sketch.py

```diff
@@ -155,59 +158,72 @@ def make_sketch(theta, beta=0.5, screen_y_scale=1.e7,
         tel = Telescope(angle=(180.*u.deg-location))
         xf, yf = tel.foot()
         tel_pos += [offset(rotate(scale(offset(tel(), (-xf, -yf)), tel_size),
-                                  location-90.*u.deg),
-                           rotate((1., 0.), location))]
+                                  rotation + location - 90.*u.deg),
+                           rotate((1., 0.), rotation + location))]
 
     # Pulsar
     ns = NeutronStar()
+    ns_x = -44.
     ns_size = 0.3
-    ns_offset = (-44., 0.)
-    ns_pos = offset(rotate(scale(ns(), ns_size), 60.*u.degree), ns_offset)
+    ns_offset = rotate((ns_x, 0.), rotation)
+    ns_pos = offset(rotate(scale(ns(), ns_size), rotation + 60.*u.degree),
+                    ns_offset)
     arrow_size = 1.5
     ns_vel = offset(rotate(offset(arrow(arrow_size), (0., 1.4)),
-                           225.*u.degree), ns_offset)
+                           rotation + 225.*u.degree), ns_offset)
+
+    # Scattering points
+    if direct is None:
+        direct = 0. in theta
 
     if direct:
         # Remove direct line of sight from input angles.
         # TODO: not strictly the same if the telescope is not at y=0!
         theta = theta[theta != 0]
-    screen_x = (1. - beta) * ns_offset[0]
-    screen_y = screen_x * np.tan(theta) * screen_y_scale
-    shortening = np.array([0.025, 0.5, 0.992])
+    lens_x = (1. - beta) * ns_x
+    lens_y = lens_x * np.tan(theta).value * screen_y_scale
+    lens_pos = []
+    for l_y in lens_y:
+        lens_pos += [rotate([lens_x, l_y], rotation)]
+
+    # Direct lines from pulsar to telescopes
+    shortening = np.array([0.025, beta, 0.992])
     tel_centers = [(t[2][0], t[3][0]) for t in tel_pos]
     p2t_pos = []
 
-    # Lines from screen to telescopes
     for tel_x, tel_y in tel_centers:
         p2t_pos += [[ns_offset[0] + shortening * (tel_x - ns_offset[0]),
                      ns_offset[1] + shortening * (tel_y - ns_offset[1])]]
 
     # Lines from pulsar to screen to telescopes
     p2s2t_pos = []
-    for s_y in screen_y:
+    for lens in lens_pos:
         p2s2t = []
         for p2t in p2t_pos:
             p = tuple(p.copy() for p in p2t)
-            p[0][1] = screen_x
-            p[1][1] = s_y
-            p[1][0] = s_y * shortening[0]/shortening[1]
+            p[0][1] = lens[0]
+            p[1][1] = lens[1]
+            p[0][0] = ns_offset[0] \
```

Comment:
More nit-picking: `\` is generally discouraged. Instead, put parentheses around the expression, like
```
p[0][0] = (ns_offset[0]
           + shortening[0]/shortening[1]
           * (lens[0] - ns_offset[0]))
```